kore/src/Kore/Builtin/Map/Map.hs

```diff
@@ -247,3 +255,45 @@ asTermLike builtin =
         TermLike variable
     element (key, MapValue value) =
         mkApplySymbol elementSymbol [key, value]
+
+externalize1 ::
+    InternalVariable variable =>
+    InternalMap Key (TermLike variable) ->
+    Either (TermLike variable)
+        (Recursive.Base
+            (Syntax.Pattern variable Attribute.Null)
+            (Free
+                (Recursive.Base (Syntax.Pattern variable Attribute.Null))
+                (TermLike variable)
+            )
+        )
+externalize1 builtin =
+    AssocComm.externalize1
+        (AssocComm.UnitSymbol unitSymbol)
+        (AssocComm.ConcatSymbol concatSymbol)
+        (map concreteElement (Map.toAscList concreteElements))
+        (element . unwrapElement <$> elementsWithVariables)
+        filteredMaps
+  where
+    filteredMaps = filter (not . isEmptyMap) opaque
```

Comment:
Why filter the opaque terms? I fear too much modification between the internal and external structure will mask bugs in the future.

Reply:
That was the behaviour of the `asTermLike` prior to my changes, but I'll remove that if you think that would be better.

Reply:
Yes, for unparsing I think that would be much better.